src/product_logging/framework.rs

```diff
@@ -244,6 +244,151 @@ log4j.appender.FILE.layout=org.apache.log4j.xml.XMLLayout
+///
+/// let mut cm_builder = ConfigMapBuilder::new();
+/// cm_builder.metadata(ObjectMetaBuilder::default().build());
+///
+/// if let Some(ContainerLogConfig {
+///     choice: Some(ContainerLogConfigChoice::Automatic(log_config)),
+/// }) = logging.containers.get(&Container::MyProduct)
+/// {
+///     cm_builder.add_data(
+///         LOG4J2_CONFIG_FILE,
+///         product_logging::framework::create_log4j2_config(
+///             &format!("{STACKABLE_LOG_DIR}/my-product"),
+///             MY_PRODUCT_LOG_FILE,
+///             MAX_LOG_FILE_SIZE_IN_MIB,
+///             CONSOLE_CONVERSION_PATTERN,
+///             log_config,
+///         ),
+///     );
+/// }
+///
+/// cm_builder.build().unwrap();
+/// ```
+pub fn create_log4j2_config(
+    log_dir: &str,
+    log_file: &str,
+    max_size_in_mib: u32,
+    console_conversion_pattern: &str,
+    config: &AutomaticContainerLogConfig,
+) -> String {
+    let number_of_archived_log_files = 1;
+
+    let logger_names = config
+        .loggers
+        .iter()
+        .filter(|(name, _)| name.as_str() != AutomaticContainerLogConfig::ROOT_LOGGER)
+        .map(|(name, _)| name.escape_default().to_string())
+        .collect::<Vec<String>>()
+        .join(", ");
+    let loggers = if logger_names.is_empty() {
+        "".to_string()
+    } else {
+        format!("loggers = {}", logger_names)
+    };
+    let logger_configs = config
+        .loggers
+        .iter()
+        .filter(|(name, _)| name.as_str() != AutomaticContainerLogConfig::ROOT_LOGGER)
+        .map(|(name, logger_config)| {
+            format!(
+                "logger.{name}.name = {name}\nlogger.{name}.level = {level}\n",
+                name = name.escape_default(),
+                level = logger_config.level.to_log4j_literal(),
+            )
+        })
+        .collect::<String>();
+
+    format!(
+        r#"
+appenders = FILE, CONSOLE
+
+appender.CONSOLE.type = Console
+appender.CONSOLE.name = CONSOLE
+appender.CONSOLE.target = SYSTEM_ERR
+appender.CONSOLE.layout.type = PatternLayout
+appender.CONSOLE.layout.pattern = {console_conversion_pattern}
+appender.CONSOLE.filter.threshold.type = ThresholdFilter
+appender.CONSOLE.filter.threshold.level = {console_log_level}
+
+appender.FILE.type = RollingFile
+appender.FILE.name = FILE
+appender.FILE.fileName = {log_dir}/{log_file}
+appender.FILE.filePattern = {log_dir}/{log_file}.%i
+appender.FILE.layout.type = XMLLayout
+appender.FILE.policies.type = Policies
+appender.FILE.policies.size.type = SizeBasedTriggeringPolicy
+appender.FILE.policies.size.size = {max_log_file_size_in_mib}MB
+appender.FILE.strategy.type = DefaultRolloverStrategy
+appender.FILE.strategy.max = {number_of_archived_log_files}
+appender.FILE.filter.threshold.type = ThresholdFilter
+appender.FILE.filter.threshold.level = {file_log_level}
+{loggers}
+{logger_configs}
+rootLogger.level={root_log_level}
+rootLogger.appenderRefs = root
+rootLogger.appenderRef.root.ref = FILE"#,
```

Comment:
The reference to the console appender is missing, isn't it?

```suggestion
rootLogger.appenderRefs = CONSOLE, FILE
rootLogger.appenderRef.CONSOLE.ref = CONSOLE
rootLogger.appenderRef.FILE.ref = FILE"#,
```